docs/RFCS/freeze.md

```diff
@@ -0,0 +1,90 @@
+- Feature Name: Data/network freeze
+- Status: draft
+- Start Date: 2016-02-18
+- Authors: Ben Darnell
+- RFC PR:
+- Cockroach Issue:
+
+
+# Summary
+
+This RFC outlines the plan for freezing our data formats and network
+protocols.
+
+# Motivation
+
+We currently make backwards-incompatible changes to data formats
+without providing any means to upgrade without data loss. This will
+need to stop before beta for obvious reasons.
```

Comment:
Have we considered not making this guarantee for the beta release? This would allow us to, for example, provide a dump/restore tool instead of being forced to write migration logic for all migrations ever. This could be useful if one of the more difficult-to-change data formats needs a revision. It's obvious that we need this guarantee as we near a stable release, but we may be biting off a large support burden here for a relatively small number of users who may not be using it seriously.